### PR TITLE
Finish faster conversions

### PR DESCRIPTION
**Pull request by janherich**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

Implements faster (~3x) `realm->clj` conversions by leveraging schemas.

Also as side effect of doing this, tedious manual conversions of lists (primitives or objects) are no longer required, so functions like `fix-map` and `fix-map->vec` can be thrown away and overall amount of code is decreased.

I also discovered that deleting transport records has been done incorrectly and fixed that, so it's quite possible that couple of group chat bugs (like being able to receive messages from chat which user left) is fixed as well.

### Testing notes (optional):
Broad regression testing with focus on persistence, eq if things done behave correctly after app restart.

status: ready

### CODE REVIEW

File: src/status_im/data_store/realm/core.cljs [hunk 1]
```diff
@@ -156,74 +143,56 @@
...
-  (some-> results single internal-convert))
+(defn single-clj [results schema-name]
+  (-> results single (realm-obj->clj schema-name)))
 
-(defn- get-schema-by-name [opts]
-  (->> opts
-       (mapv (fn [{:keys [name] :as schema}]
-               [(keyword name) schema]))
-       (into {})))
+(defn all-clj [results schema-name]
```
Comment (jeluard) [inline src/status_im/data_store/realm/core.cljs]: Can we have some doc? :)

File: src/status_im/data_store/realm/transport.cljs [hunk 2]
```diff
@@ -4,8 +4,7 @@
 
 (defn get-all
   []
-  (-> (realm/get-all @realm/account-realm :transport)
-      realm/js-object->clj))
+  (realm/all-clj (realm/get-all @realm/account-realm :transport) :transport))
```
Comment (jeluard) [inline src/status_im/data_store/realm/transport.cljs:7]: Do we ever want `get-all` without `all-clj` ?

**Review by jeluard — COMMENTED** [review]
Great job! Definitively an improvement.
It would make sense to have some tests for the realm conversion part, if possible.

File: src/status_im/data_store/realm/transport.cljs (hunk 2 above)
Comment (janherich) [reply src/status_im/data_store/realm/transport.cljs:7]: Potentially yes, if one wants for example to select all realm entities of some type (let's say messages) and read one property from each into set or whatever. Makes sense to not force anybody to do the whole object realm->clj conversion just for that.

File: src/status_im/data_store/realm/core.cljs (hunk 1 above)
Comment (janherich) [reply src/status_im/data_store/realm/core.cljs]: Sure, I will add them :)

**Review by jeluard — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]

**Review by rasom — APPROVED** [review]

**Comment by rasom** [comment]
btw @janherich are these numbers from iOS simulator? I got this:
```clojure
(time 
 (let [account-obj (aget (.objects base-realm "account") 0)]
   (dotimes [_ 1000]
     (realm-obj->clj account-obj :account))))
"Elapsed time: 566.000000 msecs"
```

**Review by yenda — APPROVED** [review]

**Comment by rasom** [comment]
So it's like 3 times difference:
```clojure
(time 
 (let [account-obj (aget (.objects base-realm "account") 0)]
   (dotimes [i 1000]
     (js-object->clj account-obj))))
"Elapsed time: 1504.000000 msecs"
```
i'm just wondering how numbers are that different

**Comment by janherich** [comment]
@rasom the number posted above are from genymotion virtual device(Android 6.0), I didn't try it with iOS simulator (as it's generally very fast and android is more performance sensitive), I will try it now and post the results.

**Comment by janherich** [comment]
@rasom I now tried on iOS simulator and it's indeed only ~3 times faster, I'm puzzled why I got that result on android... Still worth it, but I'm changing the description :)

**Comment by rasom** [comment]
The reason why i checked it is that the difference is really huge and tbh it doesn't make sense why it would be 200x.  On slow android device results are
before:
```clojure
(time 
 (let [account-obj (aget (.objects base-realm "account") 0)]
   (dotimes [i 1000]
     (js-object->clj account-obj))))
"Elapsed time: 36572.000000 msecs"
```
after:
```clojure
(time 
 (let [account-obj (aget (.objects base-realm "account") 0)]
   (dotimes [i 1000]
     (doall (realm-obj->clj account-obj :account)))))
"Elapsed time: 10303.000000 msecs"
```
@janherich i suppose you run it on empty database or something

**Comment by janherich** [comment]
@rasom Probably was some weird condition, like compilation going on (I checked it in repl one immediately after another), and I was over-optimistic and didn't check it multiple times :)

**Comment by Serhy** [comment]
There are no regressions with the changes made here, but also no side-effect-fixes as well: Group chat issues with the members leaving the chat are kept. 
Well anyway, it's good to be merged!

**Comment by janherich** [comment]
@Serhy Thanks, as it was not primary purpose of the PR (group chat issue), I would still merge it anyway, will take care of the group chat issues in another outstanding PR

**Pull request merged**